### PR TITLE
[ServiceNow] DLS documentation

### PR DESCRIPTION
**Pull request by parthpuri-elastic**
## Part of https://github.com/elastic/connectors/issues/1819

Add Documentation of ServiceNow DLS

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/document-level-security/SERVICENOW.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+##### `Enable document level security`
+
+Toggle to enable [document level security (DLS)](https://www.elastic.co/guide/en/enterprise-search/master/dls.html). When enabled:
+- Full syncs will fetch access control lists for each document and store them in the `_allow_access_control` field.
+- Access control syncs will fetch users' access control lists and store them in a separate index.
+
+#### Limitations
+
+- ServiceNow connector does not support ServiceNow scripted and conditional permissions.
```
Comment (seanstory) [inline docs/document-level-security/SERVICENOW.md:23]: Should we document what we do support? Sharing the mapping of roles to resources that we're using?

Comment (parthpuri-elastic) [reply docs/document-level-security/SERVICENOW.md:35]: sure, added

File: docs/document-level-security/SERVICENOW.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.
+
+## Document level security
+
+Document level security (DLS) enables you to restrict access to documents based on a user's permissions. This feature is available by default for the ServiceNow connector.
+
+Refer to [document level security](https://www.elastic.co/guide/en/enterprise-search/master/dls.html) for more information.
+
+ServiceNow connector support ACL roles for document level security. For default services, connectors use the below roles to find users who have access to documents.
```
Comment (seanstory) [inline docs/document-level-security/SERVICENOW.md:11]:
```suggestion
ServiceNow connector supports ACL roles for document level security. For default services, connectors use the below roles to find users who have access to documents.
```

File: docs/document-level-security/SERVICENOW.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+| Service        | Roles                                                             |
+|----------------|-------------------------------------------------------------------|
+| User           | admin                                                             |
+| Incident       | admin, sn_incident_read, ml_report_user, ml_admin, itil           |
+| Requested Item | admin, sn_request_read, asset, atf_test_designer, atf_test_admin  |
+| Knowledge      | admin, knowledge, knowledge_manager, knowledge_admin              |
+| Change Request | admin, sn_change_read, itil                                       |
+
+For services other than default, connector iterate over access controls with read operations and find the respective roles of the services.
```
Comment (seanstory) [inline docs/document-level-security/SERVICENOW.md:21]:
```suggestion
For services other than the defaults, the connector iterates over access controls with read operations and finds the respective roles for those services.
```

Do we support non-default services though? I thought we only supported these. Or is this an Advanced Sync Rules edge case?

Comment (parthpuri-elastic) [reply docs/document-level-security/SERVICENOW.md]: Yes, we are supporting non-default services as well.

**Review by seanstory — APPROVED** [review]

**Pull request merged**